src/engine/Collision/Body.ts

```diff
@@ -108,139 +124,184 @@ export class Body {
+   */
+  @obsolete({ message: 'Will be removed in v0.24.0', alternateMethod: 'Body.usePolygonCollider' })
+  public usePolygonCollision(points: Vector[], center: Vector = Vector.Zero) {
+    this.usePolygonCollider(points, center);
+  }
+
+  /**
+   * Sets up a [[Circle|circle collision geometry]] with a specified radius in pixels.
    *
    * By default, the box is center is at (0, 0) which means it is centered around the actors anchor.
    */
-  public useCircleCollision(radius?: number, center: Vector = Vector.Zero) {
+  public useCircleCollider(radius?: number, center: Vector = Vector.Zero): Collider {
     if (!radius) {
-      radius = this.actor.getWidth() / 2;
+      radius = this._actor.getWidth() / 2;
     }
-    this.collisionArea = new CircleArea({
-      body: this,
+    this.collider.shape = new Circle({
+      collider: this.collider,
       radius: radius,
       pos: center
     });
-    this.moi = this.collisionArea.getMomentOfInertia() || this.moi;
+    this.collider.inertia = this.collider.shape.getInertia() || this.collider.inertia;
+    return this.collider;
+  }
+
+  /**
+   * @obsolete Body.useCircleCollision will be removed in v0.24.0, use [[Body.useCircleCollider]]
+   */
+  @obsolete({ message: 'Will be removed in v0.24.0', alternateMethod: 'Body.useCircleCollider' })
+  public useCircleCollision(radius?: number, center: Vector = Vector.Zero) {
+    this.useCircleCollider(radius, center);
   }
 
   /**
-   * Sets up an [[EdgeArea|edge collision]] with a start point and an end point relative to the anchor of the associated actor
+   * Sets up an [[Edge|edge collision geometry]] with a start point and an end point relative to the anchor of the associated actor
    * of this physics body.
    *
    * By default, the box is center is at (0, 0) which means it is centered around the actors anchor.
    */
-  public useEdgeCollision(begin: Vector, end: Vector) {
-    this.collisionArea = new EdgeArea({
+  public useEdgeCollider(begin: Vector, end: Vector) {
+    this.collider.shape = new Edge({
+      collider: this.collider,
       begin: begin,
-      end: end,
-      body: this
+      end: end
     });
 
-    this.moi = this.collisionArea.getMomentOfInertia() || this.moi;
+    this.collider.inertia = this.collider.shape.getInertia() || this.collider.inertia;
   }
 
-  /* istanbul ignore next */
-  public debugDraw(ctx: CanvasRenderingContext2D) {
-    // Draw motion vectors
-    if (Physics.showMotionVectors) {
-      DrawUtil.vector(ctx, Color.Yellow, this.pos, this.acc.add(Physics.acc));
-      DrawUtil.vector(ctx, Color.Red, this.pos, this.vel);
-      DrawUtil.point(ctx, Color.Red, this.pos);
-    }
-
-    if (Physics.showBounds) {
-      this.getBounds().debugDraw(ctx, Color.Yellow);
-    }
-
-    if (Physics.showArea) {
-      this.collisionArea.debugDraw(ctx, Color.Green);
-    }
+  private _wireColliderEventsToActor() {
```

Comment:
Add TODO around this for ECS to remove, eventually events will stay local to the thing they are about